File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) - 1

        while (l != r and numbers[l] + numbers[r] != target):
            result = numbers[l] + numbers[r]
            if result > target:
                r -= 1
            elif result < target:
                l += 1


        
        return [l+1, r+1]

File: Data Structures & Algorithms/two-integer-sum/submission-2.go
func twoSum(nums []int, target int) []int {
    // val, index
    m := make(map[int]int)   
    for i := 0; i < len(nums); i++ {
        v, prs := m[target-nums[i]]
        if prs {
            return []int{v, i}
        }
        m[nums[i]] = i
    }
    return []int{0,0}
}
